math/fibonacci_fast.cpp

```diff
@@ -19,28 +19,28 @@
 #include <cstdio>
 #include <iostream>
 
-/** maximum number that can be computed - The result after 93 cannot be stored
- * in a `uint64_t` data type. */
-const uint64_t MAX = 93;
-
-/** Array of computed fibonacci numbers */
-uint64_t f[MAX] = {0};
-
 /** Algorithm */
 uint64_t fib(uint64_t n) {
+    // maximum number that can be computed - The result after 93 cannot be
+    // stored in a `uint64_t` data type.
+    static const uint64_t MAX = 93;
+
+    // Array of computed fibonacci numbers */
+    static uint64_t numbers[MAX] = {0};
```

Comment:
```suggestion
static uint64_t f0 = 1, f1=1;
```

Reply:
We shall use another name, it's too short.

Reply:
What about `variable0`, and `variable1`?
Or `variable1`, `variable2`.

Reply:
the variables are declared static within the function. Hence, there wouldn't be any problem. The issue happens when declaring *global* variables with short names like that.

Reply:
In summary, what I am suggesting is something like this:
```cpp
uint64_t fib(uint64_t n) {
    static uint64_t f1 = 1, f2 = 1;  // using static keyword will retain the values of f1 and f2 for the next function call.
    
    if (n <= 2)
        return f2;
    if (n >= 93) {
        std::err << "Cannot compute for n>93 due to limit of 64-bit integers\n";
        return 0;
    }

    uint64_t temp = f2;
    f2 += f1;
    f1 = temp;

    return f2;
}
```

Reply:
I created a `large_fibonacci.cpp` on similar lines with this but it can generate any n^th Fibonacci number - even 1000^th.